scripts/build/watch.js

```diff
@@ -0,0 +1,92 @@
+const Gaze = require('gaze').Gaze;
+const { spawn } = require('child_process');
+const rootdir = __dirname + '/../../';
+
+const watchers = [];
+
+const GAZE_OPTIONS = {
+  interval: 1000 // how often the target should be polled in milliseconds
+};
+
+const configs = [
+  // admin css
+  { 
+    files: [ 'admin/src/css/**/*' ],
+    task: 'exec:less',
+  },
+
+  // admin-js
+  {
+    files: [ 'admin/src/js/**/*', 'shared-libs/*/src/**/*' ],
+    task: 'exec:browserify-admin',
+  },
+
+  // admin-templates
+  {
+    files: [ 'admin/src/templates/**/*' ],
+    task: 'exec:compile-admin-templates',
+  },
+
+  // webapp-js
+  // instead of watching the source files which are watched separately, watch the build folder and upload on rebuild
+  {
+    files: [ 'api/build/static/webapp/**/*', '!api/build/static/webapp/service-worker.js' ],
+    task: 'update-service-worker',
+  },
+
+  // api-public-files
+  {
+    files: [ 'api/src/public/**/*' ],
```

Comment:
I'm making a single change in a file in this folder, which ends up copying all files, and then subsequently triggers another task 4 times. Is there a way we can debounce them? 
```
/home/diana/projects/medic-second/api/src/public/privacy-policy/script.js updated...
Running "exec:copy-api-resources" (exec) task
Done.

/home/diana/projects/medic-second/api/build/static/webapp/robots.txt updated...
/home/diana/projects/medic-second/api/build/static/webapp/manifest.json updated...
/home/diana/projects/medic-second/api/build/static/webapp/manifest.appcache updated...
/home/diana/projects/medic-second/api/build/static/webapp/appcache-upgrade.html updated...
Running "update-service-worker" task
Running "update-service-worker" task
Running "update-service-worker" task
Running "update-service-worker" task
Done.
Done.
Done.
Done.
```

Reply:
Yes there is... but it looks like the default is 500 so it's not working: https://github.com/shama/gaze/blob/19b87085606e80e02e5ca05ea942cf8a34732b59/lib/gaze.js#L37

I'll dig in to this and see if I can work out why...

Reply:
I think we'd need this PR: https://github.com/shama/gaze/pull/258/files

Reply:
Ultimately the problem is... you modify 1 file, which calls a brute force grunt task (yet to be converted) which copies all files from one folder to another (even though you only modified one). This then triggers the watcher for each file copied (I'm not sure why it only fires 4 times but maybe it maxes out somewhere?) which builds the service worker each time.

Options...

1. Use the patch linked above. This introduces a lag (but we could make it like 10ms or something tiny), but means we're now on a fork of gaze which sucks (even though it's not being updated anyway...).
2. Add the equivalent delay in our watch.js which means we're main of gaze.
3. Make the copy commands smarter - when the gaze event fires it tells you which file was updated. If we copy just this file then updating one file will only copy one file and only trigger one "update-service-worker" task. This is good because there's no point copying files that aren't updated anyway, but it's a lot more maintenance because we need to maintain the brute force version (for fresh builds) as well as the targeted version (for incremental builds).

I'm leaning towards option 2 because then I make the change in one place and everything just works. What do you think?

Reply:
I think option 2 sounds good. it's cheapest to try out anyway :)

Reply:
Done.